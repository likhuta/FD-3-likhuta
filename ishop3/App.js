"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

import MyComponent from './components/MyComponent';


let titleOfTable= require('./titleOfTable.json');
let infoForTable= require('./infoForTable.json');

ReactDOM.render(
  <MyComponent 
    titleOfTable={titleOfTable}
    infoForTable={infoForTable}
  />
  , document.getElementById('container') 
);

