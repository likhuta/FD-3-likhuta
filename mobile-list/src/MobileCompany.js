import React from 'react';
import './MobileCompany.css';
import MobileClient from './MobileClient'
import FormField from './FormField'
import {channelEvents} from './events';

class MobileCompany extends React.PureComponent {

  state={
    nameCompany:this.props.name,
    clients:this.props.clients,
    workModelFormField:1,
    // 1 - форма не отображается
    // 2 выбрали элемент для редактирования
    workModelSort:1,
    //1 -  показывает всех клиентов
    chooseClient:'',
  }

  saveClient='saveClient';
  chooseClient='chooseClient';
  deleteClient='deleteClient';
  addClient='addClient';

  componentDidMount =()=>{
    channelEvents.addListener(this.saveClient, this.EOsaveClient)
    channelEvents.addListener(this.chooseClient,this.EOchooseClient)
    channelEvents.addListener(this.deleteClient, this.EOdeleteClient)
  }

  componentWillMount =()=>{
    channelEvents.removeListener(this.saveClient, this.EOsaveClient)
    channelEvents.removeListener(this.chooseClient, this.EOchooseClient)
    channelEvents.removeListener(this.deleteClient, this.EOdeleteClient)

  }

  EOsaveClient=(param)=>{
    console.log('EOsaveClient', param);
    // сохранить нового/измененного клиента
    // нажата кнопка сохранить
  }

  EOchooseClient=(id)=>{
    console.log('EOchooseClient', id)
    // редактировали клиента
  }

  EOdeleteClient=(id)=>{
    console.log('EOdeleteClient', id)
    // нажата кнопка удалить клиента из списка
  }

  EOaddClient=()=>{
    channelEvents.emit(this.addClient,)
  }

  render(){

    console.log("MobileCompany render");
    console.log(this.state.clients);

    var clientsCode=<table> 
     {  this.state.clients.map(client=>
      <MobileClient key={client.id} info={client} />
    )}
    </table>
    
   
    


    return(
      <div className={'out_border'}>
        <input type={'button'} value={'Velcom'} />
        <input type={'button'} value={'MTS'} />
        <p>Компания {this.state.nameCompany}</p>
        <hr/>

        <input type={'button'} value={'Все'} />
        <input type={'button'} value={'Активные'} />
        <input type={'button'} value={'Заблокированные'} />
        <hr/>

       { clientsCode}

       <input type={'button'} value={'Добавить клиента'} onClick={this.EOaddClient} />
       <br/>
       <br/>


      <FormField>

      </FormField>





      </div>

      
    )
  }

}

export default MobileCompany;