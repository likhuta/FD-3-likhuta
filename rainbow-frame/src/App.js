import React, { Component } from 'react';
import './App.css';
import RainbowFrame from  './RainbowFrame'
import BR from  './BR'


class App extends Component {
  render() {
   let colors = ['red','orange', 'yellow','green', '#00BFFF', 'blue', 'purple'];
   let text="первый<br>второй<br/>третий<br />последний";
   
  return (
    <span> 
    <RainbowFrame colors={colors}>
    
      Hello!
   
    </RainbowFrame>
    {
      // второе задание 
    }
    <BR text={text} />
    </span>
  );
  }
}

export default App;
