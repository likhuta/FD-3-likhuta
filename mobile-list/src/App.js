import React from 'react';
import './App.css';
import MobileCompany from './MobileCompany';


class App extends React.PureComponent {



 

  
  render() {

    let companyName='Velcom';
    let clientsArr=[ 
    {id:101, nameClient:'Иван', serNameClient:'Краснов', secondNameClient:'Сергеевич' , balance:200}, 
   {id:105, nameClient:'Сергей', serNameClient:'Зайцев', secondNameClient:'Владимирович' , balance:250}, 
    {id:1215, nameClient:'Мария', serNameClient:'Иванова', secondNameClient:'Александровна' , balance:-10}, 
   {id:589, nameClient:'Борис', serNameClient:'Смирнов', secondNameClient:'Геннадьевич' , balance:50}, 

    ]; 

    return (
    <MobileCompany 
    name={companyName}
    clients={clientsArr}
     /> 

    )

  }
  
}

export default App;
