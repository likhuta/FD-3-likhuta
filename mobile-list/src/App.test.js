import React from 'react';
//import ReactDOM from 'react-dom';
import App from './App';
import MobileCompany from './MobileCompany';

// его нужно установить
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils';


it('renders without crashing', () => {
  const component = renderer.create(
    <App />
  );
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  /*

  const buttonElem = component.root.find( el => el.props.id == 2 ); 
  buttonElem.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  ReactTestUtils.Simulate.click(buttonElem,{id:2})
  */

 const value = 2;
 const onClick = jest.fn();
 const wrapper = shallow(
     <MobileCompany />
 );

 expect(wrapper).toMatchSnapshot();

 wrapper.find({ id: 2 }).simulate('click', {
 target: { value },
});

expect(onClick).toBeCalledWith(value);

});
