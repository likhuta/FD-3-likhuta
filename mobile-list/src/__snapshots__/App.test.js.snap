// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="out_border"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <p>
    Компания 
    Velcom
  </p>
  <hr />
  <input
    id={1}
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    id={2}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    id={3}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <tbody>
      <tr>
        <th>
          Иван
        </th>
        <th>
          Краснов
        </th>
        <th>
          Сергеевич
        </th>
        <th>
          200
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Сергей
        </th>
        <th>
          Зайцев
        </th>
        <th>
          Владимирович
        </th>
        <th>
          250
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Мария
        </th>
        <th>
          Иванова
        </th>
        <th>
          Александровна
        </th>
        <th>
          -10
        </th>
        <th
          className="blocked_client"
        >
          blocked_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Борис
        </th>
        <th>
          Смирнов
        </th>
        <th>
          Геннадьевич
        </th>
        <th>
          50
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
  <br />
</div>
`;

exports[`renders without crashing 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clients": Array [
        Object {
          "balance": 200,
          "id": 101,
          "nameClient": "Иван",
          "secondNameClient": "Сергеевич",
          "serNameClient": "Краснов",
        },
        Object {
          "balance": 250,
          "id": 105,
          "nameClient": "Сергей",
          "secondNameClient": "Владимирович",
          "serNameClient": "Зайцев",
        },
        Object {
          "balance": -10,
          "id": 1215,
          "nameClient": "Мария",
          "secondNameClient": "Александровна",
          "serNameClient": "Иванова",
        },
        Object {
          "balance": 50,
          "id": 589,
          "nameClient": "Борис",
          "secondNameClient": "Геннадьевич",
          "serNameClient": "Смирнов",
        },
      ],
      "name": "Velcom",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clients": Array [
          Object {
            "balance": 200,
            "id": 101,
            "nameClient": "Иван",
            "secondNameClient": "Сергеевич",
            "serNameClient": "Краснов",
          },
          Object {
            "balance": 250,
            "id": 105,
            "nameClient": "Сергей",
            "secondNameClient": "Владимирович",
            "serNameClient": "Зайцев",
          },
          Object {
            "balance": -10,
            "id": 1215,
            "nameClient": "Мария",
            "secondNameClient": "Александровна",
            "serNameClient": "Иванова",
          },
          Object {
            "balance": 50,
            "id": 589,
            "nameClient": "Борис",
            "secondNameClient": "Геннадьевич",
            "serNameClient": "Смирнов",
          },
        ],
        "name": "Velcom",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 3`] = `
<div
  className="out_border"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <p>
    Компания 
    Velcom
  </p>
  <hr />
  <input
    id={1}
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    id={2}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    id={3}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <tbody>
      <tr>
        <th>
          Иван
        </th>
        <th>
          Краснов
        </th>
        <th>
          Сергеевич
        </th>
        <th>
          200
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Сергей
        </th>
        <th>
          Зайцев
        </th>
        <th>
          Владимирович
        </th>
        <th>
          250
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Мария
        </th>
        <th>
          Иванова
        </th>
        <th>
          Александровна
        </th>
        <th>
          -10
        </th>
        <th
          className="blocked_client"
        >
          blocked_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th>
          Борис
        </th>
        <th>
          Смирнов
        </th>
        <th>
          Геннадьевич
        </th>
        <th>
          50
        </th>
        <th
          className="active_client"
        >
          active_client
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировть"
          />
        </th>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </th>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
  <br />
</div>
`;
